### PR TITLE
chore: remove dependencies and fix typescript warnings

### PR DESCRIPTION
**Pull request by BugGambit**
Changes:
- Removing dependencies url, query-string and is-buffer. They are either trivial or have native node/browser replacements.
- Fix some typescript warnings.
- Increase timeout on an integration test (flaky).

### CODE REVIEW

File: packages/core/src/authFlows/__tests__/auth.spec.ts [hunk 1]
```diff
@@ -246,7 +246,7 @@ describe('Cognite Auth', () => {
         });
         expect(spiedLocationAssign).toBeCalledTimes(1);
         expect(spiedLocationAssign.mock.calls[0][0]).toMatchInlineSnapshot(
-          `"https://example.com/login/redirect?errorRedirectUrl=https%3A%2F%2Ferror-redirect.com&project=my-tenant&redirectUrl=https%3A%2F%2Fredirect.com"`
+          `"https://example.com/login/redirect?project=my-tenant&redirectUrl=https%3A%2F%2Fredirect.com&errorRedirectUrl=https%3A%2F%2Ferror-redirect.com"`
```
Comment (BugGambit) [inline packages/core/src/authFlows/__tests__/auth.spec.ts:249]: They are the same, just query params changed order.

File: packages/stable/src/__tests__/api/documentsApi.int.spec.ts [hunk 2]
```diff
@@ -173,23 +173,27 @@ describe('Documents integration test', () => {
...
+
+        const pdfPrefix = [0x25, 0x50, 0x44, 0x46, 0x2d]; // %PDF-
+        expect(resp.byteLength).toBeGreaterThan(pdfPrefix.length);
+        const frontSlice = resp.slice(0, pdfPrefix.length);
+        expect(frontSlice.byteLength).toStrictEqual(pdfPrefix.length);
+        const match = Buffer.from(frontSlice, 0).equals(
+          Buffer.from(pdfPrefix, 0)
+        );
+        expect(match).toBe(true);
+      },
```
Comment (BugGambit) [inline packages/stable/src/__tests__/api/documentsApi.int.spec.ts:194]: Indent caused by added timeout

File: packages/core/src/authFlows/__tests__/auth.spec.ts (hunk 1 above)
Comment (cyriaquebrousse) [reply packages/core/src/authFlows/__tests__/auth.spec.ts:249]: why is the change needed then? 🤔

Comment (BugGambit) [reply packages/core/src/authFlows/__tests__/auth.spec.ts:249]: Using URLSearchParams vs query-string which use different ordering rules

Comment (BugGambit) [reply packages/core/src/authFlows/__tests__/auth.spec.ts:249]: And the matching here is on string level, it doesn't know this represents an URL

File: packages/core/src/utils.ts [hunk 3]
```diff
@@ -39,6 +38,21 @@ export function isBrowser() {
   );
 }
 
+function isBuffer(obj: any): boolean {
```
Comment (cyriaquebrousse) [inline packages/core/src/utils.ts:41]: does this essentially check whether `obj` is a byte array? My JS is a bit rusty.

**Review by danlevings — APPROVED** [review]
sgtm, looks like just a tidy up 👌

File: packages/core/src/authFlows/__tests__/auth.spec.ts (hunk 1 above)
Comment (cyriaquebrousse) [reply packages/core/src/authFlows/__tests__/auth.spec.ts:249]: Right!

File: packages/core/src/utils.ts (hunk 3 above)
Comment (BugGambit) [reply packages/core/src/utils.ts:41]: No, just a nodejs Buffer.
The reason we can't use Buffer.isBuffer is that this code must be compatible with browsers where `Buffer` isn't a concept

**Review by cyriaquebrousse — APPROVED** [review]

**Pull request merged**